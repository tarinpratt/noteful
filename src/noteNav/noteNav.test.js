import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom';
import NoteNav from './noteNav'

describe(`NoteNav component`, () => {
  const props = {
    folder: {
      "name": "Important"
    }
  }

  it('renders Note Navigation', () => {
    const wrapper = shallow(<NoteNav />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a title with folder name', () => {
    const h3 = shallow(<NoteNav {...props} />)
      .find('.NotePageNav__folder-name')
    expect(toJson(h3)).toMatchSnapshot()
  })

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
    <NoteNav />
    </BrowserRouter>, 
  div
  )
  ReactDOM.unmountComponentAtNode(div);
})

})