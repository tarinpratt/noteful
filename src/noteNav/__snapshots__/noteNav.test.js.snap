// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteNav component renders Note Navigation 1`] = `
<div
  className="noteNav"
>
  <button
    className="backButton"
    onClick={[Function]}
  >
    Back
  </button>
</div>
`;

exports[`NoteNav component renders a title with folder name 1`] = `null`;
